torch/_dynamo/variables/constant.py

```diff
@@ -87,19 +87,11 @@ def call_method(
             ).call_method(tx, name, args, kwargs)
 
         if any([isinstance(x, DynamicShapeVariable) for x in args]):
-            # NOTE! DANGER! THIS ONLY WORKS FOR COMMUTATIVE OPS
-            # we are relying on add to have arg[0] be a DynamicShapeVariable
-            # because we are in ConstantVariable land
-            # This transforms
-            # constant + dynamic
-            # into
-            # dynamic + constant
-            # Which already has infra built for writing to the graph
-            if name == "__add__":
-                assert len(args) == 1
-                return args[0].call_method(tx, name, [self], {})
-            # Unfortunate constant
-            return super(ConstantVariable, self).call_method(tx, name, args, kwargs)
+            # Promote to DynamicShapeVariable for operations involving dynamic shapes.
+            return variables.DynamicShapeVariable(
+                self.as_proxy(), self.value
+            ).call_method(tx, name, args, kwargs)
```

Comment:
Ah, you hit a gap in our implementation! So this is a little unsupported - ConstantVariable has no real proxy  (self.as_proxy() on a constant just gives us the real value) - meanwhile, DynamicShapeVariable has assumptions around proxy being both real and there - I would instead suggest either:

1) making a new proxy from scratch to track operations, with a ConstantSource - there is some unfortunate code in register_attr_or_module that might serve as reference

2) Make a dead-end variable - something like NumericalVariable - that is proxy-less, like ConstantSource, but knows how to do numerical operations on Dyn and Consts together (worse option)

cc: @ezyang  for input too

Reply:
What's wrong with the proxy just being the Python literal? You can still trace with it (as long as there's a proxy somewhere, which there is)